### ISSUE TITLE
Routing issue, Højby Denmark

### ISSUE THREAD

**Issue by elgaard** [osmandapp/Osmand#8152]
Using offline maps, Dec 1. 2019, but also many older maps.
routing profile: car
Osmand 3.5.5

![hojby](https://user-images.githubusercontent.com/779040/71549131-9e9c8300-29b8-11ea-86f8-ea5a3fbe6f5b.jpg)

The obviouw route is through "Tværvej"

OSRM and Grasshopper finds the best route:
https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=55.90945%2C11.58998%3B55.91348%2C11.59373#map=17/55.91148/11.59184

**Comment by elgaard** [comment]
Selecting "fuel-efficent way" does make  Osmand find the shorter way, 376m insted of 512m

I have changed Tværvej from residential to unclassified.
The Danish name does indicate that it is more of a shortcut than a residential road.

This may help Osmand when new maps become available.

**Comment by hvdwolf** [comment]
You should not have done this. Those streets (also the ones around it) are tagged residential for a reason: because they are residential.
Simply because OsmAnds priority mechanism for streets did not do the correct thing, you change an OSM street parameter that works for any navi app to something that OsmAnd can handle.
You insert incorrect data into OSM to correct a routing bug in OsmAnd (I quickly checked Magic Erarth and that one routes correct as well).
Never change the map to suit a bug in one navi app. I also know what tvaer mentions in Danish, but that is not a good argument to change that street parameter.
Please reverse it.

**Comment by elgaard** [comment]
I did not change it just to help Osmand routing (and I do not know if it will make a difference for Osmand routing). It is road that I travel regularly and looking at it again I assessed that it us more than a residential road. It is a natural way to travel north/south through Højby.

There are residential homes on Tværvej.
But there are also residential homes on Lyngvej and Højby Hovedgade tagged as tertiary.
Klokkelynges is tagged as unclassified.

**Comment by vshcherb** [comment]
It is a question of some attributes of https://www.openstreetmap.org/way/107138388 (unclassified), we treat it with 20% penalty comparing to tertiary and it is why it is causing the trouble. It is probably bad in that situation but it is better in other parts of the world. Sorry I don't see a solution for now


			<select value="0.9" t="highway" v="tertiary"/>
			<select value="0.85" t="highway" v="tertiary_link"/>
			
			<select value="0.7" t="highway" v="unclassified"/>

**Comment by Zero3** [comment]
@elgaard Yes, the default OsmAnd routing config does weird stuff like this. See this related comment I just made in another GitHub issue about the same problem: https://github.com/osmandapp/OsmAnd/issues/2625#issuecomment-904131236. My work-in-progress `routing.xml` for Denmark that I mentioned finds the "correct" route through Tværvej, so it is indeed a `routing.xml` issue.

**Comment by scaidermern** [comment]
> I did not change it just to help Osmand routing (and I do not know if it will make a difference for Osmand routing). It is road that I travel regularly and looking at it again I assessed that it us more than a residential road. It is a natural way to travel north/south through Højby.

Then you should use tertiary, not unclassified.
